src/main/java/first/repository/StudentRepository.java

```diff
@@ -79,15 +73,13 @@ public void update(Student student) {
             preparedStatement.execute();
             System.out.println( "студент обновлен в базе" );
         } catch (SQLException e) {
-            e.printStackTrace();
+            System.err.println(e);
         }
     }
 
     public Student getStudentById(int id) {
         try {
-            preparedStatement = connection.prepareStatement(
-                    "SELECT * FROM Students where id=" + id );
-
+            preparedStatement = connection.prepareStatement("SELECT * FROM Students where id=" + id );
```

Comment:
You used preparedStatement not correctly. In this case you should write: connection.prepareStatement("SELECT * FROM Students WHERE id=?", id);

Also, please, look around and find using all prepared statements and check

Reply:
Code is not working after that...

Reply:
So, I forgot this, try to 
preparedStatement = connection.prepareStatement("SELECT * FROM Students WHERE id=?");
preaparedStatement.setInt(id);